cpp/src/io/parquet/page_data.cu

```diff
@@ -912,16 +912,14 @@ static __device__ bool setupLocalPageInfo(page_state_s *const s,
   int chunk_idx;
 
   // Fetch page info
-  // NOTE: Assumes that sizeof(PageInfo) <= 256 (and is padded to 4 bytes)
-  if (t < sizeof(PageInfo) / sizeof(uint32_t)) {
-    reinterpret_cast<uint32_t *>(&s->page)[t] = reinterpret_cast<const uint32_t *>(p)[t];
-  }
+  if (t == 0) s->page = *p;
   __syncthreads();
+
   if (s->page.flags & PAGEINFO_FLAGS_DICTIONARY) { return false; }
   // Fetch column chunk info
   chunk_idx = s->page.chunk_idx;
+  // TO DO, need to replace, currently has some race condition
```

Comment:
ideally every such load should be succeeded by a syncthreads

Reply:
There is a syncthreads below at line 938, and as this parameter is not being used in the middle thought it wouldn't be of any use to call one more syncthreads.